Create test traceability description for Application Connector Manager (#78)
kyma-project/application-connector-manager: docs/contributor/01-11-ci-cd.md

> # CI/CD

Reviewer: Can you change the file name to `04-10-ci-cd.md`?

| Name                                                                                                                                                  | Required | Platform   | Description                                                                                                                                                                                             |
|-------------------------------------------------------------------------------------------------------------------------------------------------------|----------|------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| [`pre-application-connector-manager-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L1059)       | true     | prow       | Builds Application Connector Manager's image and pushes it to the `dev` registry.                                                                                                                       |

> | [`pre-application-connector-manager-presubmit-scanner`](https://github.com/kyma-project/test-infra/prow/jobs/test-infra/presubmit-scanner.yaml#L556)  | true     | prow       | Run Gitleaks presubmit scanner to detect committed sensitive data for Application Connector Manager.                                                                                                    |

Reviewer: This link is broken. Maybe it should be this one - https://github.com/kyma-project/test-infra/blob/main/prow/jobs/test-infra/presubmit-scanner.yaml ?

> | [`pre-application-connector-manager-presubmit-scanner`](https://github.com/kyma-project/test-infra/prow/jobs/test-infra/presubmit-scanner.yaml#L556)  | true     | prow       | Run Gitleaks presubmit scanner to detect committed sensitive data for Application Connector Manager.                                                                                                    |

Reviewer: ```suggestion
| [`pre-application-connector-manager-presubmit-scanner`](https://github.com/kyma-project/test-infra/prow/jobs/test-infra/presubmit-scanner.yaml#L556)  | true     | prow       | Runs the Gitleaks presubmit scanner to detect committed sensitive data for Application Connector Manager.                                                                                                    |
```

| [`pre-application-connector-module-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L1031)        | true     | prow       | Builds module's OCI image and pushes it to the `dev` artifact registry. Renders ModuleTemplate for the Application Connector module that allows for manual integration tests against Lifecycle Manager. |

> | [`pre-main-application-connector-manager-verify`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L1093) | true     | prow       | Installs Application Connector Manager, not using Lifecycle Manager, and applies the sample Application Connector CR on a k3d cluster. Executes smoke integration test of Application Connector         |

Reviewer: ```suggestion
| [`pre-main-application-connector-manager-verify`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L1093) | true     | prow       | Installs Application Connector Manager, not using Lifecycle Manager, and applies the sample Application Connector CR on a k3d cluster. Executes smoke integration test of Application Connector.         |
```

> | [`golangci-lint `](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/golangci-lint.yaml)                      | true     | GH Actions | Is responsible for linting, static code analysis.                                                                                                                                                       |

Reviewer: ```suggestion
| [`golangci-lint `](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/golangci-lint.yaml)                      | true     | GH Actions | Is responsible for linting and static code analysis.                                                                                                                                                       |
```

| [`Run unit tests`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/run-tests.yaml)                          | true     | GH Actions | Executes basic create/update/delete functional tests of Application Connector Manager's reconciliation logic.                                                                                           |

> | [`Run integration tests`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/run-validation.yaml)              | true     | GH Actions | Installs Application Connector Manager, not using Lifecycle Manager, and applies the sample Application Connector CR on a k3d cluster. Executes smoke integration test of Application Connector         |

Reviewer: ```suggestion
| [`Run integration tests`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/run-validation.yaml)              | true     | GH Actions | Installs Application Connector Manager, not using Lifecycle Manager, and applies the sample Application Connector CR on a k3d cluster. Executes smoke integration test of Application Connector.         |
```

> | [`Run vuln check`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/run-vuln-check.yaml)                     | true     | GH Actions | Executes [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) tool on Application Connector Manager code to detect known vulnerabilities.                                                |

Reviewer: ```suggestion
| [`Run vuln check`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/run-vuln-check.yaml)                     | true     | GH Actions | Executes the [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) tool on the Application Connector Manager code to detect known vulnerabilities.                                                |
```

| [`post-application-connector-manager-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L1073)                       | prow       | Re-builds manager's image and pushes it into the `prod` registry.                                                                                                                                                                                                    |
| [`post-application-connector-module-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L1007)                        | prow       | Re-builds the module's OCI image and pushes it to the `prod` artifact registry.                                                                                                                                                                                      |

> | [`post-application-connector-module-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L1007)                        | prow       | Re-builds the module's OCI image and pushes it to the `prod` artifact registry.                                                                                                                                                                                      |

Reviewer: ```suggestion
                                                                                                                                                                                    
```
Duplicate of the previous entry.

| [`Run integration tests`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/run-validation.yaml)                               | GH Actions | Installs Application Connector Manager, using Lifecyle Manager, applies Kyma CR and enables Application Connector module on a k3d cluster. Executes smoke integration test of Application Connector.                                                                 |
| [`Run unit tests`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/run-tests.yaml)                                           | GH Actions | Executes basic create/update/delete functional tests of Application Connector Manager's reconciliation logic.                                                                                                                                                        |

> | [`golangci-lint`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/golangci-lint.yaml)                                        | GH Actions | Is responsible for linting, static code analysis.                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| [`golangci-lint`](https://github.com/kyma-project/application-connector-manager/blob/main/.github/workflows/golangci-lint.yaml)                                        | GH Actions | Is responsible for linting and static code analysis.                                                                                                                                                                                                                    |
```
